api/graphql/reservations/recurring_reservation_serializers/create_serializers.py

```diff
@@ -0,0 +1,124 @@
+from rest_framework import serializers
+
+from api.graphql.base_serializers import PrimaryKeySerializer
+from api.graphql.choice_fields import ChoiceIntegerField
+from api.graphql.primary_key_fields import IntegerPrimaryKeyField
+from api.graphql.validating_list_field import ValidatingListField
+from api.graphql.validation_errors import ValidationErrorCodes, ValidationErrorWithCode
+from reservation_units.models import ReservationUnit
+from reservations.models import AbilityGroup, AgeGroup, RecurringReservation
+from tilavarauspalvelu.utils.commons import WEEKDAYS
+
+
+class RecurringReservationCreateSerializer(PrimaryKeySerializer):
+    reservation_unit_pk = IntegerPrimaryKeyField(
+        queryset=ReservationUnit.objects.all(),
+        source="reservation_unit",
+        allow_null=False,
+    )
+    age_group_pk = IntegerPrimaryKeyField(
+        queryset=AgeGroup.objects.all(),
+        source="age_group",
+        allow_null=True,
+        required=False,
+    )
+    ability_group_pk = IntegerPrimaryKeyField(
+        queryset=AbilityGroup.objects.all(),
+        source="ability_group",
+        allow_null=True,
+        required=False,
+    )
+
+    weekdays = ValidatingListField(
+        child=ChoiceIntegerField(choices=WEEKDAYS.CHOICES, allow_null=False),
+        allow_empty=False,
+        required=True,
+        help_text="List of weekdays which days the reservations occurs",
+    )
+
+    name = serializers.CharField(required=False, default="")
+    begin_time = serializers.TimeField(
+        required=True, help_text="Time when reservations begins."
+    )
+    end_time = serializers.TimeField(
+        required=True, help_text="Time when reservations ends."
+    )
+    begin_date = serializers.DateField(
+        required=True, help_text="Date when first reservation begins."
+    )
+    end_date = serializers.DateField(
+        required=True, help_text="Date when last reservation begins."
+    )
+    interval = serializers.IntegerField(required=True)
+
+    class Meta:
+        model = RecurringReservation
+        fields = [
+            "pk",
+            "user",
+            "name",
+            "reservation_unit_pk",
+            "age_group_pk",
+            "ability_group_pk",
+            "interval",
+            "weekdays",
+            "begin_time",
+            "end_time",
+            "begin_date",
+            "end_date",
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields["user"].readonly = True
+
+    def validate(self, data):
+        begin = data.get("begin_time", getattr(self.instance, "begin_time", None))
+        end = data.get("end_time", getattr(self.instance, "end_time", None))
+
+        if end < begin:
```

Comment:
This isn't probably a real world situation, but is it possible that we have a valid time that is invalid because of timezones?

Let's say we have a reservation from 02:00 - 04:00 @ UTC+3. If time is stored in UTC, this becomes 23:00 - 01:00. I didn't really get a grasp on how timezones are handled with `TimeField` but can this become a problem now that time is stored separately without date information? If I recall correctly, we just had an similar issue somewhere else.

Reply:
Very valid point, those begin and end date and times could be datetimes 🤔  Or perhaps we can just define a timezone to the recurring model. Nevertheless a ponder.

Reply:
We talked about this and decided that we use the configured timezone in settings. So no need for changing anything here. So the UI will need know the timezone in which the backend is working in.